### PR TITLE
Feature/feature perfil medic

### PR DESCRIPTION
**Pull request by enrichp**

### CODE REVIEW

File: app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrUpdatePetWash.java [hunk 1]
```diff
@@ -55,7 +55,7 @@ public void shouldUpdateWashDate() throws WashAlreadyExistingException,
         trNewPetWash.setPet(linux);
         trNewPetWash.setWash(originalWash);
         trNewPetWash.execute();
-        originalWash.setWashDate(DateTime.Builder.build(2011, 5, 27, 11, 15, 1));
+        originalWash.setDateTime(DateTime.Builder.build(2011, 5, 27, 11, 15, 1));
```
Comment (santidrj) [inline app/src/test/java/org/pesmypetcare/mypetcare/controllers/TestTrUpdatePetWash.java:58]: Codacy Issue found: ['15' is a magic number.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Coverage increased per file
===========================
+ app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  11
         

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Wash.java  -6
         

Complexity increasing per file
==============================
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetIllness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetIllness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Illness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubMedicalProfileManagerService.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMedicalProfile.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Vaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetVaccinations.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java  3
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227&bid=18112672)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetWashing.java [hunk 2]
```diff
@@ -343,7 +343,7 @@ private void initializeWashComponent(Event wash) {
      */
     private void initializeButtonLogic(Wash wash, MaterialButton washButton) {
         String washButtonText = getString(R.string.wash) + SPACE + wash.getWashDescription() + EOL
-                + getString(R.string.from_date) + SPACE + wash.getDateTime() + EOL + getString(R.string.wash_duration)
+                + getString(R.string.from_date) + SPACE + wash.getDateTime().toDateString() + EOL + getString(R.string.wash_duration)
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetWashing.java:346]: Codacy Issue found: [Line is longer than 120 characters (found 133).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetVaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMedicalProfile.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubMedicalProfileManagerService.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Vaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Illness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetIllness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetVaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetVaccinations.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227&bid=18112672)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Vaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Illness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetVaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetVaccinations.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMedicalProfile.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetIllness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubMedicalProfileManagerService.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetIllness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetVaccination.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227&bid=18112672)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Event.java  -6
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Wash.java  -6
         

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Illness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMedicalProfile.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetVaccinations.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetIllness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetVaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Vaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubMedicalProfileManagerService.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227&bid=18112672)

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java [hunk 3]
```diff
@@ -222,11 +224,74 @@ void addExercise(Pet pet, String exerciseName, String exerciseDescription, DateT
...
      * @param txtDescription The exercise description
      * @param startExerciseDateTime The exercise start DateTime
-     * @param endExerciseDateTime The exercise end DateTime
+     * @param endExerciseDateTime The exercise end DateTime-
      */
     void updateExercise(Pet pet, String txtExerciseName, String txtDescription, DateTime originalStartDateTime,
                         DateTime startExerciseDateTime, DateTime endExerciseDateTime);
 
+    /**
+     * Add an vaccination to the pet.
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java]: It's "a vaccication" not "an vaccination"

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java [hunk 4]
```diff
@@ -222,11 +224,74 @@ void addExercise(Pet pet, String exerciseName, String exerciseDescription, DateT
...
+     * @param pet The pet to whom the vaccination has to be added
+     * @param vaccinationDescription The description of the vaccination that has to be added to the pet
+     * @param vaccinationDate The date of the vaccination that has to be added to the pet
+     */
+    void addPetVaccination(Pet pet, String vaccinationDescription, DateTime vaccinationDate);
+
+    /**
+     * Updates an vaccination of the pet.
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java]: It's "a vaccication" not "an vaccination"

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java [hunk 5]
```diff
@@ -222,11 +224,74 @@ void addExercise(Pet pet, String exerciseName, String exerciseDescription, DateT
...
+     * @param pet The pet from whom we have to update the vaccination
+     * @param vaccination The vaccination with the body updated
+     * @param newDate The new date of the vaccination
+     * @param updatesDate A boolean that indicates whether the date has to be updated or not
+     */
+    void updatePetVaccination(Pet pet, Vaccination vaccination, String newDate, boolean updatesDate);
+
+    /**
+     * Deletes an vaccination from the given pet.
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java]: It's "a vaccication" not "an vaccination"

File: app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java [hunk 6]
```diff
@@ -222,11 +224,74 @@ void addExercise(Pet pet, String exerciseName, String exerciseDescription, DateT
...
+    void deletePetVaccination(Pet pet, Vaccination vaccination);
+
+    /**
+     * Obtains all vaccinations from a given pet.
+     * @param pet The pet from where we have to obtain all the vaccinations
+     */
+    void obtainAllPetVaccinations(Pet pet);
+
+    /**
+     * Add an pet illness to the pet.
```
Comment (santidrj) [inline app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetCommunication.java]: It's "a pet" not "an pet"

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+
+    /**
+     * Method responsible for creating the transaction to delete a illness from a pet.
+     * @return The transaction to delete a illness from a pet
+     */
+    public static TrDeletePetIllness createTrDeletePetIllness() {
+        return new TrDeletePetIllness(new MedicalProfileManagerAdapter());
+    }
+
+    public static TrAddNewPetIllness createTrAddNewIllness() {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java:65]: You should comment these methods

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Getter of the result of the transaction.
+     * @return A list containing all the illnesses of the pet
+     */
+    public List<Illness> getResult() {
+        return result;
+    }
+
+    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java:56]: You should comment this methods

File: app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+
+    /**
+     * Getter of the result of the transaction.
+     * @return A list containing all the vaccinations of the pet
+     */
+    public List<Vaccination> getResult() {
+        return result;
+    }
+
+    public void execute() throws NotPetOwnerException, ExecutionException, InterruptedException {
```
Comment (AlbertPG) [inline app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java:56]: You should comment this methods as well

**Review by AlbertPG — COMMENTED** [review]
You should comment these methods

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMedicalProfile.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Vaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetVaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetVaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Illness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubMedicalProfileManagerService.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetVaccinations.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetVaccination.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227&bid=18112672)

**Review by santidrj — CHANGES_REQUESTED** [review]
:trollface:

**Review by AlbertPG — DISMISSED** [review]
Okey, I approve the changes

**Review by santidrj — DISMISSED** [review]
:+1:

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetVaccination.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetVaccinations.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetVaccination.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrDeletePetIllness.java  3
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetVaccination.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetVaccinations.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrUpdatePetIllness.java  4
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Illness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/features/pets/Vaccination.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/services/StubMedicalProfileManagerService.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrObtainAllPetIllness.java  1
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetVaccination.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/MedicalProfileControllersFactory.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrAddNewPetIllness.java  3
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/activities/fragments/infopet/InfoPetMedicalProfile.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrAddNewPetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TrObtainAllPetIllness.java  4
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrDeletePetIllness.java  1
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
- app/src/test/java/org/pesmypetcare/mypetcare/controllers/medicalprofile/TestTrUpdatePetVaccination.java  1
         

Clones added
============
- app/src/main/java/org/pesmypetcare/mypetcare/services/MedicalProfileManagerAdapter.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5526227&bid=18112672)

**Review by AlbertPG — APPROVED** [review]
Okey

**Review by santidrj — APPROVED** [review]
:+1:

**Pull request merged**